README
NAME
    Acme::CPANAuthors::Austrian - We are Austrian CPAN authors

VERSION
    version 1.100710

SYNOPSIS
       use Acme::CPANAuthors;
       use Acme::CPANAuthors::Austrian;

       my $authors = Acme::CPANAuthors->new('Austrian');

       my $number   = $authors->count;
       my @ids      = $authors->id;
       my @distros  = $authors->distributions('MARCEL');
       my $url      = $authors->avatar_url('DOMM');
       my $kwalitee = $authors->kwalitee('GARGAMEL');

DESCRIPTION
    This class provides a hash of Pause ID/name of Austrian CPAN authors.

MAINTENANCE
    If you are an Austrian CPAN author and are not listed here, please mail
    me.

SEE ALSO
    Acme::CPANAuthors - Main class to manipulate this one.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org/Public/Dist/Display.html?Name=Acme-CPANAuthors-Austr
    ian>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Acme-CPANAuthors-Austrian/>.

    The development version lives at
    <http://github.com/hanekomu/Acme-CPANAuthors-Austrian/>. Instead of
    sending patches, please fork this project using the standard git and
    github infrastructure.

AUTHOR
      Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2008 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Acme-CPANAuthors-Austrian

1.100710  2010-03-12 15:00:38 Europe/Vienna
    - converted distribution to Dist::Zilla style

0.10 Tue 2010.02.09 23:54:50 CET (Marcel Gruenauer <marcel@cpan.org>)
     - releng

0.09 Sun 2010.02.07 22:26:55 CET (Marcel Gruenauer <marcel@cpan.org>)
     - added author LANTI

0.08 Mon 2009.11.16 15:05:06 CET (Marcel Gruenauer <marcel@cpan.org>)
     - added author KALEX
     - repo maintenance

0.07 (Marcel Gruenauer <marcel@cpan.org>)
     - added author NUFFIN (by Yuval's request)

0.06  2008-09-14T13:21:15Z (Marcel Gruenauer <marcel@cpan.org>)
     - added author RURBAN

0.05  2008-09-13T12:09:38Z (Marcel Gruenauer <marcel@cpan.org>)
     - fixed module pod abstract

0.04  2008-09-12T20:40:37Z (Marcel Gruenauer <marcel@cpan.org>)
     - added authors ANDK, AREIBENS

0.03  2008-09-12T16:43:25Z (Marcel Gruenauer <marcel@cpan.org>)
     - added authors DRRHO, NINE

0.02  2008-09-12T16:33:12Z (Marcel Gruenauer <marcel@cpan.org>)
     - added authors PEPL, ZEYA

0.01  2008-09-12T08:28:44Z (Marcel Gruenauer <marcel@cpan.org>)
     - original version


inc/Test/Compile.pm
#line 1
package Test::Compile;

use warnings;
use strict;
use Test::Builder;
use File::Spec;
use UNIVERSAL::require;


our $VERSION = '0.09';


my $Test = Test::Builder->new;


sub import {
    my $self = shift;
    my $caller = caller;

   for my $func (qw(
       pm_file_ok pl_file_ok all_pm_files all_pl_files all_pm_files_ok
       all_pl_files_ok
       )) {

        no strict 'refs';
        *{$caller."::".$func} = \&$func;
    }

    $Test->exported_to($caller);
    $Test->plan(@_);
}


sub pm_file_ok {
    my $file = shift;
    my $name = @_ ? shift : "Compile test for $file";

    if (!-f $file) {
        $Test->ok(0, $name);
        $Test->diag("$file does not exist");
        return;
    }

    my $module = $file;
    $module =~ s!^(blib/)?lib/!!;
    $module =~ s!/!::!g;
    $module =~ s/\.pm$//;

    my $ok = 1;
    $module->use;
    $ok = 0 if $@;

    my $diag = '';
    unless ($ok) {
        $diag = "couldn't use $module ($file): $@";
    }

    $Test->ok($ok, $name);
    $Test->diag($diag) unless $ok;
    $ok;
}


sub pl_file_ok {
    my $file = shift;
    my $name = @_ ? shift : "Compile test for $file";

    # don't "use Devel::CheckOS" because Test::Compile is included by
    # Module::Install::StandardTests, and we don't want to have to ship
    # Devel::CheckOS with M::I::T as well.

    if (Devel::CheckOS->require) {

        # Exclude VMS because $^X doesn't work. In general perl is a symlink to
        # perlx.y.z but VMS stores symlinks differently...

        unless (Devel::CheckOS::os_is('OSFeatures::POSIXShellRedirection') and
                Devel::CheckOS::os_isnt('VMS')) {

            $Test->skip('Test not compatible with your OS');
            return;
        }
    }

    unless (-f $file) {
        $Test->ok(0, $name);
        $Test->diag("$file does not exist");
        return;
    }

    my $out = `$^X -cw $file 2>&1`;

    if ($?) {
        $Test->ok(0, 'Script does not compile');
        $Test->diag($out);
        return;
    } else {
        $Test->ok(1);
        return 1;
    }
}


sub all_pm_files_ok {
    my @files = @_ ? @_ : all_pm_files();

    $Test->plan(tests => scalar @files);

    my $ok = 1;
    for (@files) {
        pm_file_ok($_) or undef $ok;
    }
    $ok;
}


sub all_pl_files_ok {
    my @files = @_ ? @_ : all_pl_files();

    $Test->plan(tests => scalar @files);

    my $ok = 1;
    for (@files) {
        pl_file_ok($_) or undef $ok;
    }
    $ok;
}


sub all_pm_files {
    my @queue = @_ ? @_ : _pm_starting_points();
    my @pm;

    while (@queue) {
        my $file = shift @queue;
        if (-d $file) {
            local *DH;
            opendir DH, $file or next;
            my @newfiles = readdir DH;
            closedir DH;

            @newfiles = File::Spec->no_upwards(@newfiles);
            @newfiles = grep { $_ ne "CVS" && $_ ne ".svn" } @newfiles;

            for my $newfile (@newfiles) {
                my $filename = File::Spec->catfile($file, $newfile);
                if (-f $filename) {
                    push @queue, $filename;
                } else {
                    push @queue, File::Spec->catdir($file, $newfile);
                }
            }
        }
        if (-f $file) {
            push @pm, $file if $file =~ /\.pm$/;
        }
    }
    @pm;
}


sub all_pl_files {
    my @queue = @_ ? @_ : _pl_starting_points();
    my @pl;

    while (@queue) {
        my $file = shift @queue;
        if (-d $file) {
            local *DH;
            opendir DH, $file or next;
            my @newfiles = readdir DH;
            closedir DH;

            @newfiles = File::Spec->no_upwards(@newfiles);
            @newfiles = grep { $_ ne "CVS" && $_ ne ".svn" } @newfiles;

            for my $newfile (@newfiles) {
                my $filename = File::Spec->catfile($file, $newfile);
                if (-f $filename) {
                    push @queue, $filename;
                } else {
                    push @queue, File::Spec->catdir($file, $newfile);
                }
            }
        }
        if (-f $file) {
            # Only accept files with no extension or extension .pl
            push @pl, $file if $file =~ /(?:^[^.]+$|\.pl$)/;
        }
    }
    @pl;
}


sub _pm_starting_points {
    return 'blib' if -e 'blib';
    return 'lib';
}


sub _pl_starting_points {
    return 'script' if -e 'script';
    return 'bin' if -e 'bin';
}


1;


__END__



#line 400



lib/Acme/CPANAuthors/Austrian.pm
use 5.006;
use strict;
use warnings;

package Acme::CPANAuthors::Austrian;
our $VERSION = '1.100710';
# ABSTRACT: We are Austrian CPAN authors

use Acme::CPANAuthors::Register (
    ANDK     => 'Andreas J. König',
    AREIBENS => 'Alfred Reibenschuh',
    DOMM     => 'Thomas Klausner',
    DRRHO    => 'Robert Barta',
    FLORIAN  => 'Florian Helmberger',
    GARGAMEL => 'Karlheinz Zöchling',
    KALEX    => 'Alexander Keusch',
    LAMMEL   => 'Roland Lammel',
    LANTI    => 'Ingo Lantschner',
    MARCEL   => 'Marcel Gruenauer == hanekomu',
    MAROS    => 'Maroš Kollár',
    NINE     => 'Stefan Seifert',
    NUFFIN   => 'Yuval Kogman',
    PEPL     => 'Michael Kröll',
    RGIERSIG => 'Roland Giersig',
    RURBAN   => 'Reini Urban',
    ZEYA     => 'Hansjörg Pehofer',
);
1;


__END__
=pod

=head1 NAME

Acme::CPANAuthors::Austrian - We are Austrian CPAN authors

=head1 VERSION

version 1.100710

=head1 SYNOPSIS

   use Acme::CPANAuthors;
   use Acme::CPANAuthors::Austrian;

   my $authors = Acme::CPANAuthors->new('Austrian');

   my $number   = $authors->count;
   my @ids      = $authors->id;
   my @distros  = $authors->distributions('MARCEL');
   my $url      = $authors->avatar_url('DOMM');
   my $kwalitee = $authors->kwalitee('GARGAMEL');

=head1 DESCRIPTION

This class provides a hash of Pause ID/name of Austrian CPAN authors.

=head1 MAINTENANCE

If you are an Austrian CPAN author and are not listed here, please mail me.

=head1 SEE ALSO

L<Acme::CPANAuthors> - Main class to manipulate this one.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Acme-CPANAuthors-Austrian>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Acme-CPANAuthors-Austrian/>.

The development version lives at
L<http://github.com/hanekomu/Acme-CPANAuthors-Austrian/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2008 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;

 BEGIN { require 5.006; } 

use ExtUtils::MakeMaker 6.11;



my %WriteMakefileArgs = (
                       'test' => {
                                   'TESTS' => 't/*.t'
                                 },
                       'NAME' => 'Acme::CPANAuthors::Austrian',
                       'DISTNAME' => 'Acme-CPANAuthors-Austrian',
                       'CONFIGURE_REQUIRES' => {
                                                 'ExtUtils::MakeMaker' => '6.11'
                                               },
                       'AUTHOR' => 'Marcel Gruenauer <marcel@cpan.org>',
                       'BUILD_REQUIRES' => {},
                       'ABSTRACT' => 'We are Austrian CPAN authors',
                       'EXE_FILES' => [],
                       'VERSION' => '1.100710',
                       'PREREQ_PM' => {
                                        'Scalar::Util' => '0',
                                        'English' => '0',
                                        'Acme::CPANAuthors::Register' => '0',
                                        'Carp' => '0',
                                        'Test::More' => '0',
                                        'Test::Differences' => '0',
                                        'File::Find' => '0',
                                        'Acme::CPANAuthors' => '0',
                                        'File::Temp' => '0'
                                      },
                       'LICENSE' => 'perl'
                     );


delete $WriteMakefileArgs{LICENSE}
  unless eval { ExtUtils::MakeMaker->VERSION(6.31) };

WriteMakefile(%WriteMakefileArgs);



